### PR TITLE
Use RHTAP to build image-controller and external-secrets

### PR DESCRIPTION
**Pull request by sbose78**
- [x]  Workspaces rhtap-build, rhtap-gitops, rhtap-servicerelease have been created
- [x] Setting up of the Application and Components have been done via the UI
- [x] Release CRs created on the clusters directly, but also being added to the gitops repo as part of this PR.
- [x] All these needs to land on the rh cluster in production.
- [x] rhtap-gitops workspace has ended up on the multi-tenant cluster by mistake, need to move it.
- [x] Enable pipelines-as-code for the image-controller after disabling the same on https://github.com/redhat-appstudio/infra-deployments/pull/2045
- [x] Enable rhtap-servicerelease to release external secrets.

### CODE REVIEW

File: argo-cd-apps/base/ci-workspaces/ci-workspaces.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+kind: ApplicationSet
+metadata:
+  name: ci-workspaces
+spec:
+  generators:
+    - merge:
+        mergeKeys:
+          - nameNormalized
+        generators:
+          - clusters:
```
Comment (gbenhaim) [inline argo-cd-apps/base/ci-workspaces/ci-workspaces.yaml:11]: you would need something like this for selecting only the production RH cluster https://github.com/redhat-appstudio/infra-deployments/blob/main/argo-cd-apps/base/tekton-ci/tekton-ci.yaml#L16

**Review by gbenhaim — CHANGES_REQUESTED** [review]

File: ci/teams/rhtap-servicerelease/release-build.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: ReleasePlanAdmission
+metadata:
+  name: rhtap-build-rpa
```
Comment (scoheb) [inline ci/teams/rhtap-servicerelease/release-build.yaml:4]: we'd also need a corresponding `rhtap-gitops-rpa`

Comment (sbose78) [reply ci/teams/rhtap-servicerelease/release-build.yaml:4]: +1, I was waiting to move the workspace to the rh cluster and also get a successful build out.

File: ci/teams/rhtap-gitops/release.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: ReleasePlan
+metadata:
+  name: release-to-quay
+  namespace: rhtap-gitops-tenant
+  labels:
+    release.appstudio.openshift.io/auto-release: 'true'
```
Comment (scoheb) [inline ci/teams/rhtap-gitops/release.yaml:7]: we also need this:

`release.appstudio.openshift.io/standing-attribution: 'true'`

File: ci/teams/rhtap-build/release.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: ReleasePlan
+metadata:
+  name: build-release-to-quay
+  namespace: rhtap-build-tenant
+  labels:
+    release.appstudio.openshift.io/auto-release: 'true'
```
Comment (scoheb) [inline ci/teams/rhtap-build/release.yaml:7]: we also need this:

`release.appstudio.openshift.io/standing-attribution: 'true'`

File: ci/teams/rhtap-gitops/release.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+spec:
+  application: external-secrets
+  displayName: build release-to-quay
```
Comment (scoheb) [inline ci/teams/rhtap-gitops/release.yaml:10]:
```suggestion
  displayName: external-secrets release-to-quay
```

File: ci/teams/rhtap-build/release.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: ReleasePlan
+metadata:
+  name: build-release-to-quay
```
Comment (scoheb) [inline ci/teams/rhtap-build/release.yaml:4]: in gitops file, you removed prefix...

```suggestion
  name: release-to-quay
```

**Review by gbenhaim — APPROVED** [review]
need to also ensure that the target namespaces exist on the RH cluster.

**Comment by sbose78** [comment]
> need to also ensure that the target namespaces exist on the RH cluster.

Yes, in fact the resources exist as well. This should be interesting to see what happens next :)

File: ci/teams/rhtap-build/release.yaml (hunk 4 above)
Comment (scoheb) [reply ci/teams/rhtap-build/release.yaml:7]: still missing

Comment (sbose78) [reply ci/teams/rhtap-build/release.yaml:7]: added 👍 Another round of review, please?

**Review by scoheb — APPROVED** [review]

**Pull request merged**